dm/dm/master/scheduler/scheduler.go

```diff
@@ -2662,3 +2663,61 @@ func (s *Scheduler) handleLoadTask(ctx context.Context, loadTaskCh <-chan ha.Loa
 		}
 	}
 }
+
+func (s *Scheduler) OperateValidationTask(expectStage pb.Stage, stCfgs map[string]map[string]config.SubTaskConfig) error {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	if !s.started.Load() {
+		return terror.ErrSchedulerNotStarted.Generate()
+	}
+	for taskName := range stCfgs {
+		release, err := s.subtaskLatch.tryAcquire(taskName)
+		if err != nil {
+			return terror.Annotatef(err, "fail to require lock for validation task")
+		}
+		defer release()
+	}
+
+	// 1. get stages of subtask's validator and update stage
+	newCfgs := make([]config.SubTaskConfig, 0)
+	validatorStages := make([]ha.Stage, 0)
+	for taskName := range stCfgs {
+		for _, cfg := range stCfgs[taskName] {
+			stageM, _, err := ha.GetValidatorStage(s.etcdCli, cfg.SourceID, cfg.Name, 0)
+			if err != nil {
+				return terror.Annotatef(err, "fail to get validator stage for task `%s` and source `%s`", cfg.Name, cfg.SourceID)
+			}
+			if v, ok := stageM[cfg.Name]; ok && v.Expect == expectStage {
+				warnInfo := fmt.Sprintf("subtask validator is already %s\n", expectStage.String())
+				s.logger.Info(
+					warnInfo,
+					zap.String("taskName", cfg.Name),
+					zap.String("source", cfg.SourceID),
+				)
+			} else {
+				if expectStage == pb.Stage_Running {
+					// don't need to update config if stopping the validator task
```

Comment:
so how validator in worker know this?

Reply:
Currently, the validator's stage is persisted in etcd. The worker observes this stage constantly and will change its stage once this field is reset by the master.